logstreams/src/main/java/io/camunda/zeebe/logstreams/impl/log/LogStorageAppender.java

```diff
@@ -178,10 +186,19 @@ public void onActorFailed() {
   }
 
   private void onWriteBufferAvailable() {
-    final BlockPeek blockPeek = new BlockPeek();
-    if (writeBufferSubscription.peekBlock(blockPeek, maxAppendBlockSize, true) > 0) {
-      appendBlock(blockPeek);
-    } else {
+    final var blockPeek = new BlockPeek();
+    final var readBytes = writeBufferSubscription.peekBlock(blockPeek, maxAppendBlockSize, true);
+
+    final var shouldAppendBlock = readBytes > 0;
```

Comment:
:wrench: NIT: might make sense to call it `canAppend` or `hasToAppend`, since there is a block we can append.

Reply:
Done